### ISSUE TITLE
ERROR for opencl version "Cannot load library (libgflags.so.2.2: cannot open shared object file: No such file or directory)

### ISSUE THREAD

**Issue by DVEfremov** [dividiti/ck-crowdsource-dnn-optimization#12]
Step to reproduce (only at my ubuntu at this moment)
* Install all required ck packages as listed at read.me 
* run crowdsource-video-experiments-on-desktop 
* select engine = Caffe OpenCL proxy 
* Press Start 
* Got error 
![image](https://cloud.githubusercontent.com/assets/22045452/22735048/9ece62f2-ee2a-11e6-8d9d-4e47affcb1ad.png)

**Comment by DVEfremov** [comment]
I've got almost the same env as @Chunosov 
```
daniil@daniil:~/CK-TOOLS/lib-caffe-bvlc-opencl-clblast-trunk-gcc-5.4.0-linux-64/install/lib$ ck show env: --tags=gflags
Env UID:         Target OS:        Bits: Name:          Version: Tags:

94ce4b84204f4b72          linux-64    64 GFlags library 2.2.0    64bits,gflags,host-os-linux-64,lib,target-os-linux-64,v2,v2.2,v2.2.0
4c5df90514a209d4   android21-arm64    64 GFlags library 2.2.0    64bits,gflags,host-os-linux-64,lib,target-os-android21-arm64,v2,v2.2,v2.2.0
2635474c8b059c1e   android21-arm64    64 GFlags library 2.1.2    64bits,gflags,host-os-linux-64,lib,target-os-android21-arm64,v2,v2.1,v2.1.2
c8987cecf81c3944 android21-arm-v7a    32 GFlags library 2.2.0    32bits,gflags,host-os-linux-64,lib,target-os-android21-arm-v7a,v2,v2.2,v2.2.0

daniil@daniil:~/CK-TOOLS/lib-caffe-bvlc-opencl-clblast-trunk-gcc-5.4.0-linux-64/install/lib$ ls -l /usr/lib/x86_64-linux-gnu/libgflags.so*
lrwxrwxrwx 1 root root     14 Sep 13  2015 /usr/lib/x86_64-linux-gnu/libgflags.so -> libgflags.so.2
lrwxrwxrwx 1 root root     18 Sep 13  2015 /usr/lib/x86_64-linux-gnu/libgflags.so.2 -> libgflags.so.2.1.2
-rw-r--r-- 1 root root 133800 Sep 13  2015 /usr/lib/x86_64-linux-gnu/libgflags.so.2.1.2
```

CPU version works fine

**Comment by DVEfremov** [comment]
I've installed same env at notebook and got  error for CPU version 
```
Start experiment for "Caffe CPU proxy" with "Caffe model (net and weights) (bvlc, googlenet)" on "ImageNet dataset (aux)"
Loading dependencies. Count: 6
Load dep lib 6 "/home/defremov/CK-TOOLS/lib-protobuf-host-3.1.0-linux-64/install/lib/libprotobuf.so"
File not found
Load dep lib 5 "/usr/lib/x86_64-linux-gnu/libhdf5.so"
Load dep lib 4 "/home/defremov/CK-TOOLS/lib-caffe-bvlc-master-cpu-trunk-gcc-4.8.4-linux-64/install/lib/libcaffe.so"
ERROR: "Cannot load library /home/defremov/CK-TOOLS/lib-caffe-bvlc-master-cpu-trunk-gcc-4.8.4-linux-64/install/lib/libcaffe.so: (libgflags.so.2.2: cannot open shared object file: No such file or directory)"
```
but it works fine using console

**Comment by DVEfremov** [comment]
Looks like /home/defremov/CK-TOOLS/lib-gflags-2.2.0-gcc-4.8.4-linux-64/install/lib
should be added to LD_LIBRARY_PATH, because there is no such file at common system env: 
```
defremov@dem:~/CK-TOOLS/lib-caffe-bvlc-master-cpu-trunk-gcc-4.8.4-linux-64/install/lib$ ldd libcaffe.so | grep gflag
	libgflags.so.2.2 => not found
```

**Comment by DVEfremov** [comment]
I've applied workaround export LD_LIBRARY_PATH with absolute path of all required libs.
For CPU  engine:
```
LD_LIBRARY_PATH=$dirname:/home/daniil/CK-TOOLS/lib-gflags-2.2.0-gcc-5.4.0-linux-64/install/lib:/home/daniil/CK-TOOLS/lib-clblast-0.10.0-gcc-5.4.0-linux-64/lib
export LD_LIBRARY_PATH
./crowdsource-video-experiments-on-desktop "$@"
``` 
and for OpenCL version:
```
LD_LIBRARY_PATH=$dirname:/home/daniil/CK-TOOLS/lib-gflags-2.2.0-gcc-5.4.0-linux-64/install/lib:/home/daniil/CK-TOOLS/lib-clblast-0.10.0-gcc-5.4.0-linux-64/lib
export LD_LIBRARY_PATH
./crowdsource-video-experiments-on-desktop "$@"
```

**Comment by dsavenko** [comment]
Had the same error with another library. My understanding is that this is because I built different dependencies with different versions of GCC (5.4 and 6.0). Should not be the issue in theory, but when I rebuilt everything from scratch with the same version of GCC, I stopped seeing this issue.

Anyway, I'm not ready to close it yet. Let's wait for @DVEfremov  and see if it's still reproducible for him.
